cmd/inferTime.go

```diff
@@ -0,0 +1,163 @@
+	buffer = buffer + fmt.Sprintf("Log position of last recorded time: %d\n", tb.logFilePosition)
+	buffer = buffer + fmt.Sprintf("Nbr of entries without time:        %d\n", tb.noTimeCount)
+	buffer = buffer + fmt.Sprintf("Computed interval:                  %d\n", tb.deltatime)
+
+	return buffer
+}
+
+//finalizeTimeGap makes the necessary checks and computation
+func (tb *InferTimeBlock) finalizeTimeGap() error {
+	var err error
+	err = nil
+
+	//Check that lastRecordedTime and nextValidTime are not null
+	nullTime := time.Time{}
+	if tb.lastRecordedTime == nullTime {
+		errMsg := "Fatal error: gap start time is empty"
+		log.Println(errMsg)
+		err = fmt.Errorf(errMsg)
+		return err
+	}
+	if tb.nextValidTime == nullTime {
+		errMsg := "Fatal error: gap end time is empty"
+		log.Println(errMsg)
+		err = fmt.Errorf(errMsg)
+		return err
+	}
+
+	//Are the two times equal?
+	if tb.nextValidTime == tb.lastRecordedTime {
+		errMsg := "Fatal error: the start and end gap times are equal"
+		log.Println(errMsg)
+		err = fmt.Errorf(errMsg)
+		return err
+	}
+
+	//Compute the time difference
+	startTime := tb.lastRecordedTime
+	endTime := tb.nextValidTime
+	diff := endTime.Sub(startTime)
+
+	//Fail if we have a negative time difference
+	if diff.Seconds() < 0 {
+		errMsg := "Fatal error: Gap start time is later than the Gap end time"
+		log.Println(errMsg)
+		return fmt.Errorf(errMsg)
+	}
+
+	//Do we have a non null noTimeCount
+	if tb.noTimeCount < 1 {
+		errMsg := fmt.Sprintf("Fatal error: invalid number of records without time (%d)\n", tb.noTimeCount)
+		log.Println(errMsg)
+		return fmt.Errorf(errMsg)
+	}
+
+	//TODO: What should we expect as logFilePosition?
+
+	//Compute the gap
+	floatInterval := diff.Seconds() / float64(tb.noTimeCount+1)
+	tb.deltatime = int(floatInterval)
+
+	return nil
+}
+
+//storeTimeGap updates an InferTimeBLock (last valid time, nbr of records without time). It returns true if we reached the end of the time gap.
+func (tb *InferTimeBlock) storeTimeGap(logline LogLine, position int) (bool, error) {
+	var err error
+	err = nil
+
+	//ActualTime is filled if a time could be found in the FLE input
+	if logline.ActualTime != "" {
+		//Are we starting a new block
+		if tb.noTimeCount == 0 {
+			tb.lastRecordedTime = convertDateTime(logline.Date + " " + logline.ActualTime)
+			tb.logFilePosition = position
+		} else {
+			// We reached the end of the gap
+			nullTime := time.Time{}
+			if tb.lastRecordedTime == nullTime {
+				errMsg := "Fatal error: gap start time is empty"
+				log.Println(errMsg)
+				err = fmt.Errorf(errMsg)
+				return false, err
+			}
+
+			tb.nextValidTime = convertDateTime(logline.Date + " " + logline.ActualTime)
+			return true, err
+		}
+	} else {
+		//Check the data is correct.
+		nullTime := time.Time{}
+		if tb.lastRecordedTime == nullTime {
+			errMsg := "Fatal error: gap start time is empty"
+			log.Println(errMsg)
+			err = fmt.Errorf(errMsg)
```

Comment:
```suggestion
			return false, errors.New("gap start time is empty")
```

When you spot an error, it's safer to return it asap. Otherwise you might end up in cases where you continue executing with your error already raised.

Reply:
implemented